src/Shape/Box.js

```diff
@@ -0,0 +1,385 @@
+            const dataObj = {};
+            dataObj.__d3plus__ = true;
+            dataObj.data = eachValue;
+            dataObj.i = index;
+            dataObj.outlier = this._outlier(d, i);
+
+            if (orient === "vertical") {
+              dataObj.x = d.x;
+              if (value < d.bottom) {
+                dataObj.y = d.y + d.height / 2 + d.first - d.bottom + value;
+                outlierData.push(dataObj);
+              }
+              else if (value > d.top) {
+                dataObj.y = d.y - d.height / 2 - d.top - d.third - value;
+                outlierData.push(dataObj);
+              }
+            }
+            else if (orient === "horizontal") {
+              dataObj.y = d.y;
+              if (value < d.bottom) {
+                dataObj.x = d.x - d.width / 2 - d.first - d.bottom - value;
+                outlierData.push(dataObj);
+              }
+              else if (value > d.top) {
+                dataObj.x = d.x + d.width / 2 + d.top - d.third + value;
+                outlierData.push(dataObj);
+              }
+            }
+          }
+
+        });
+
+        d.nested = true;
+        d.__d3plus__ = true;
+
+        return d;
+      });
+
+    // Draw box.
+    new Rect()
+      .data(filteredData)
+      .x(d => d.x)
+      .y(d => d.y)
+      .select(elem("g.d3plus-Box", {parent: this._select}).node())
+      .config(configPrep.bind(this)(this._rectConfig, "shape"))
+      .render();
+
+    // Draw median.
+    new Rect()
+      .data(filteredData)
+      .x(d => orient === "vertical" ? d.x : d.medianX)
+      .y(d => orient === "vertical" ? d.medianY : d.y)
+      .height(d => orient === "vertical" ? 1 : d.height)
+      .width(d => orient === "vertical" ? d.width : 1)
+      .select(elem("g.d3plus-Box-Median", {parent: this._select}).node())
+      .config(configPrep.bind(this)(this._medianConfig, "shape"))
+      .render();
+
+    // Draw 2 lines using Whisker class.
+    // Construct coordinates for whisker startpoints and push it to the whiskerData.
+    const whiskerData = [];
+    filteredData.forEach((d, i) => {
+      const x = d.x;
+      const y = d.y;
+      const topLength = d.top - d.third;
+      const bottomLength = d.first - d.bottom;
+
+      if (orient === "vertical") {
+        const topY = y - d.height / 2;
+        const bottomY = y + d.height / 2;
+        whiskerData.push(
+          {__d3plus__: true, data: d, i, x, y: topY, length: topLength, orient: "top"},
+          {__d3plus__: true, data: d, i, x, y: bottomY, length: bottomLength, orient: "bottom"}
+        );
+      }
+      else if (orient === "horizontal") {
+        const topX = x + d.width / 2;
+        const bottomX = x - d.width / 2;
+        whiskerData.push(
+          {__d3plus__: true, data: d, i, x: topX, y, length: topLength, orient: "right"},
+          {__d3plus__: true, data: d, i, x: bottomX, y, length: bottomLength, orient: "left"}
+        );
+      }
```

Comment:
The math to construct the whisker lengths is incorrect. `d.top` is technically the visual bottom of the whisker (y positions on a screen in pixels go from top to bottom, not bottom to top). If you run this code, the whiskers extend outside of the screen, which is incorrect because none of the values are negative. If the whiskerMode is set to "extent", then the top whisker of this first box should be 20px from the top of the screen, and the bottom whisker should be 300px from the top of the screen.

```js
var data = [
  {id: "alpha", value: 300},
  {id: "alpha", value: 20},
  {id: "alpha", value: 180},
  {id: "alpha", value: 40},
  {id: "alpha", value: 170},
  {id: "alpha", value: 125},
  {id: "alpha", value: 74},
  {id: "alpha", value: 80},
  {id: "beta", value: 180},
  {id: "beta", value: 30},
  {id: "beta", value: 120},
  {id: "beta", value: 50},
  {id: "beta", value: 140},
  {id: "beta", value: 115},
  {id: "beta", value: 14},
  {id: "beta", value: 30}
];

var area = new d3plus.Box()
  .data(data)
  .x(d => d.id === "alpha" ? 100 : 200)
  .y(d => d.value)
  .render();
```